resolwe_runtime_utils.py

```diff
@@ -275,14 +301,233 @@ def checkrc(rc, *args):
     return json.dumps(ret)
 
 
+def export_file(file_path):
+    """Prepend the given parameter with ``export``"""
+
+    if not os.path.isfile(file_path):
+        return error("Referenced file does not exist: '{}'.".format(file_path))
+
+    return "export {}".format(file_path)
+
+
+CHUNK_SIZE = 10000000  # 10 Mbytes
+
+
+class ImportedFormat:
+    """Import destination file format."""
+
+    EXTRACTED = 'extracted'
+    COMPRESSED = 'compressed'
+    BOTH = 'both'
+
+
+def import_file(
+    src,
+    file_name,
+    imported_format=ImportedFormat.BOTH,
+    progress_from=0.0,
+    progress_to=None,
+):
+    """Import file to working directory.
+
+    :param src: Source file path or URL
+    :param file_name: Source file name
+    :param imported_format: Import file format (extracted, compressed or both)
+    :param progress_from: Initial progress value
+    :param progress_to: Final progress value
+    :return: Destination file path (if extracted and compressed, extracted path given)
+    """
+
+    if progress_to is not None:
+        if not isinstance(progress_from, float) or not isinstance(progress_to, float):
+            raise ValueError("progress_from and progress_to must be float")
+
+        if progress_from < 0 or progress_from > 1:
+            raise ValueError("progress_from must be between 0 and 1")
+
+        if progress_to < 0 or progress_to > 1:
+            raise ValueError("progress_to must be between 0 and 1")
+
+        if progress_from >= progress_to:
+            raise ValueError("progress_to must be higher than progress_from")
+
+    print("Importing and compressing {}...".format(file_name))
+
+    def importGz():
+        """Import gzipped file.
+
+        The file_name must have .gz extension.
+        """
+        if imported_format != ImportedFormat.COMPRESSED:  # Extracted file required
+            with open(file_name[:-3], 'wb') as f_out, gzip.open(src, 'rb') as f_in:
+                shutil.copyfileobj(f_in, f_out, CHUNK_SIZE)
```

Comment:
How should I handle the exception?

Reply:
I suggest to handle it the same way as 5 lines below.

Reply:
Hm. That handles the case where the file name is the the same as downloaded file and we can skip the copy. This can't happen above, because the function handles uncompressed files and here we compress it (so src and file_name) are always different.

Reply:
We are looking at different sections of the code and I mean [this line](https://github.com/genialis/resolwe-runtime-utils/pull/6/files#diff-1815e39b348d47412791b8389c9ed432R368). We extract compressed `gz` file and raise `ValueError` in the linked line but we don't do that here.

Reply:
Done